components/cdc/src/delegate.rs

```diff
@@ -620,6 +632,53 @@ impl Delegate {
         if entries.is_empty() {
             return Ok(());
         }
+
+        let downstreams = self.downstreams();
+        assert!(
+            !downstreams.is_empty(),
+            "region {} miss downstream",
+            self.region_id
+        );
+
+        let mut need_filter = false;
+        for ds in downstreams {
+            if ds.filter_loop {
+                need_filter = true;
+                break;
+            }
+        }
+
+        // collect the change event cause by user write, which is `txn_source` = 0.
+        // for changefeed which only need the user write, send the `filtered`, or else,
+        // send them all.
+        let (filtered, empty) = if need_filter {
```

Comment:
What about `Option<Event>` instead of `(filtered, empty)`?

Reply:
done, PTAL